### PR TITLE
#4337 Fix eslint errors

### LINKED ISSUE #4337: Clean up linting errors
## Is your feature request related to a problem? Please describe.

It's not a big problem. But it would be nice to not have linting errors in the codebase 😸 
image

## Describe the solution you'd like

Address each error (and warnings, where possible) individually

## Implementation

TBD when investigating

## Describe alternatives you've considered

Just leaving it. It wouldn't be the end of the world 🌏 

## Additional context
N/A

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #4337 

## Change summary

- Few small changes, shouldn't have any functional impact

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Run `yarn eslint src` in the mobile repo directory, verify no errors and app still works

### Related areas to think about
N/A

### CODE REVIEW

File: src/authentication/DemoSiteRequest.js [hunk 1]
```diff
@@ -12,20 +12,14 @@ export class DemoSiteRequest {
   async createActivationURL(username, email, password, repeatPassword) {
     this.validateFields(username, email, password, repeatPassword);
 
-    let responseJson;
-    try {
```
Comment (katherine-sussol) [inline src/authentication/DemoSiteRequest.js:16]: Fixing this:
`16:5  error  Unnecessary try/catch wrapper  no-useless-catch`

File: src/database/DataTypes/Period.js [hunk 2]
```diff
@@ -65,9 +65,7 @@ export class Period extends Realm.Object {
 
   get numberOfDays() {
     return (
-      moment(this.endDate)
-        .startOf('day')
-        .diff(moment(this.startDate).startOf('day'), 'days') + 1
+      moment(this.endDate).startOf('day').diff(moment(this.startDate).startOf('day'), 'days') + 1
```
Comment (katherine-sussol) [inline src/database/DataTypes/Period.js:68]: Fixing this:
` 68:27  error  Replace `⏎········.startOf('day')⏎········` with `.startOf('day')`  prettier/prettier`

File: src/navigation/actions.js [hunk 3]
```diff
@@ -12,8 +12,7 @@ import { createRecord } from '../database/utilities';
 import { navStrings } from '../localization';
 import { ROUTES } from './constants';
 import { RootNavigator } from './RootNavigator';
-import { PrescriptionActions } from '../actions/PrescriptionActions';
-import { FinaliseActions } from '../actions/FinaliseActions';
+import { PrescriptionActions, FinaliseActions } from '../actions/index';
```
Comment (katherine-sussol) [inline src/navigation/actions.js:15]: For
`  9:1  error  Dependency cycle via ./actions:7=>../actions/FinaliseActions:16  import/no-cycle`

File: src/widgets/BreachManHappy.js [hunk 4]
```diff
@@ -225,3 +226,8 @@ export const BreachManHappy = ({ width = 100, height = 80 }) => (
     </G>
   </Svg>
 );
+
+BreachManHappy.propTypes = {
```
Comment (katherine-sussol) [inline src/widgets/BreachManHappy.js:230]: For
```
  15:34  error  'width' is missing in props validation   react/prop-types
  15:47  error  'height' is missing in props validation  react/prop-types
```

File: .eslintrc [hunk 5]
```diff
@@ -37,6 +37,7 @@
     "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
     "react/jsx-one-expression-per-line": "off",
     "react/jsx-props-no-spreading": "off",
+    "react/static-property-placement": ["warn", "property assignment"],
```
Comment (katherine-sussol) [inline .eslintrc:40]: For
`  14:5  error  'propTypes' should be declared outside the class body  react/static-property-placement`

**Review by katherine-sussol — COMMENTED** [review]
Added comments with the eslint complaints inline

File: src/widgets/withOnePress.js [hunk 6]
```diff
@@ -9,10 +9,13 @@ import React from 'react';
  */
 export const withOnePress = WrappedComponent =>
   class extends React.Component {
-    state = { hasBeenPressed: false };
-
     static propTypes = WrappedComponent.propTypes;
 
+    constructor(props) {
```
Comment (katherine-sussol) [inline src/widgets/withOnePress.js:14]: For
`  12:5  error  State initialization should be in a constructor        react/state-in-constructor`

**Review by joshxg — APPROVED** [review]
Great, thansk!

**Pull request merged**